src/core/lib/layer-manager.js

```diff
@@ -664,19 +660,22 @@ export default class LayerManager {
     } else {
       log.log(LOG_PRIORITY_LIFECYCLE_MINOR, `Matching layer is unchanged ${newLayer.id}`)();
       newLayer.lifecycle = LIFECYCLE.MATCHED;
-      newLayer.oldProps = newLayer.props;
     }
   }
 
   // Updates a single layer, cleaning all flags
-  _updateLayer(layer) {
+  _updateLayer(layer, {transferStateFromLayer} = {}) {
+    if (transferStateFromLayer) {
+      this._transferLayerState(transferStateFromLayer, layer);
```

Comment:
* For clarity, I would call `layer._transferState` directly here.
* Check for "self" inside `layer._transferState`
* Rename `LayerManager._transferLayerState` to `_logLayerMatching`.
* Figure out the best place to call _logLayerMatching. Perhaps one level up.